lib/puppet/ssl/certificate_request.rb

```diff
@@ -267,7 +267,17 @@ def extension_request_attribute(options)
 
     if options[:dns_alt_names]
       names = options[:dns_alt_names].split(/\s*,\s*/).map(&:strip) + [name]
-      names = names.sort.uniq.map {|name| "DNS:#{name}" }.join(", ")
+
+      names = names.sort.map do |name|
+        if (!name.start_with? "IP:") && (!name.start_with? "DNS:")
+          "DNS:#{name}"
+        else
+          "#{name}"
+        end
+      end
+
+      names = names.uniq.join(", ")
```

Comment:
We shouldn't assign the same variable too many times in a method. We might want to combine this and the above statement and name the first assignment `raw_names` and the second `parsed_names` or something.

Reply:
The reason I split it out into two statements was because if we sort and append the DNS prefix in the same statement, then we see the duplication issue @Magisus mentioned. Or did you mean something else? Either way, I'll fix the naming issue.

Reply:
No meant combining them, I remember there be a duplication issue, but I can't imagine how chaining those methods would cause it:
```rb
irb(main):011:0> foo = ['baba', 'titi', 'dodo', 'baba', 'papa', 'mama', 'gigi', 'mama']
=> ["baba", "titi", "dodo", "baba", "papa", "mama", "gigi", "mama"]
irb(main):012:0> foo.sort.map{|n| "PREFIX:#{n}" }.uniq.join(", ")
=> "PREFIX:baba, PREFIX:dodo, PREFIX:gigi, PREFIX:mama, PREFIX:papa, PREFIX:titi"
```

Reply:
Ah, I think the problem was that originally we were calling .uniq before we added the prefix. I'll just move it to afterwards like you did.